src/ipa_pc/mod.rs

```diff
@@ -736,6 +736,155 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
         })
     }
 
+    /// Refers to the "Multi-poly multi-point assertions" section of the "Recursive SNARKs based on Marlin" document
+    fn batch_open_individual_opening_challenges<'a>(
+        ck: &Self::CommitterKey,
+        labeled_polynomials: impl Clone + IntoIterator<Item = &'a LabeledPolynomial<G::ScalarField>>,
+        commitments: impl Clone + IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
+        query_set: &QuerySet<G::ScalarField>,
+        opening_challenges: &dyn Fn(u64) -> G::ScalarField,
```

Comment:
Unlike for ``open_individual_opening_challenges()`` it is not good design decision to hand over a function for the opening challenges (which is initialized in a secure manner by the caller), as we squeeze another random challenge internally. 

One can of course do a work around for now, and hash all the generated opening challenges plus the the new commitment to get the new query point `x` via a ``compute_random_oracle_challenge()`` call, but a cleaner way is to avoid splitting the Fiat-Shamir transform between outside and inside ``batch_open_individual_opening_challenge()``.
There are two ways to do that:

1. Pass a sponge instead, which is used to squeeze all the random challenges (and which absorbs the new commitment before squeezing the new query point `x`),
2. Implement an independent hash-based Fiat-Shamir transform inside ``batch_open_...()`` according to the following steps. 
     - We initialize our state by hashing the claim of the (batched) opening proof, state = ``compute_random_oracle_challenge(all commitments, all points, and all claimed values)``. In our setting, we can use this state as the first lambda, ``lambda_1 = state``.
    - all other challenges lambda_2, ..., lambda_n are squeezed recursively by ``lambda_i = 
       compute_random_oracle_challenge(lambda_{i-1})``.
    - The new opening point `x`  is squeezed by absorbing the new batch poly and the last lambda_n, ``x= 
       compute_random_oracle_challenge(lambda_n, C_newpoly)``.

Reply:
For now, we decided to keep it as it is, and put the following comment above the declaration of ``batch_open_individual_opening_challenges()``:
TODO: In order to implement the Fiat-Shamir transform in a "clean" manner, we need to pass a sponge instead of ``opening_challenges()``. This sponge is then used to squeeze the opening challenge(s) as well as the fresh query point `x`.

Reply:
I've put the comment to the code